libdevcrypto/SecretStore.cpp

```diff
@@ -84,36 +85,38 @@ static js::mValue upgraded(std::string const& _s)
 	return js::mValue();
 }
 
-SecretStore::SecretStore(std::string const& _path): m_path(_path)
+SecretStore::SecretStore(string const& _path): m_path(_path)
 {
 	load();
 }
 
-SecretStore::~SecretStore()
+bytes SecretStore::secret(h128 const& _uuid, function<string()> const& _pass, bool _useCache) const
 {
-}
-
-bytes SecretStore::secret(h128 const& _uuid, function<std::string()> const& _pass, bool _useCache) const
-{
-	(void)_pass;
 	auto rit = m_cached.find(_uuid);
 	if (_useCache && rit != m_cached.end())
 		return rit->second;
 	auto it = m_keys.find(_uuid);
-	if (it == m_keys.end())
-		return bytes();
-	bytes key = decrypt(it->second.first, _pass());
-	if (!key.empty())
-		m_cached[_uuid] = key;
+	bytes key;
+	if (it != m_keys.end())
+	{
+		key = decrypt(it->second.encryptedKey, _pass());
+		if (!key.empty())
+			m_cached[_uuid] = key;
+	}
 	return key;
 }
 
-h128 SecretStore::importSecret(bytes const& _s, std::string const& _pass)
+h128 SecretStore::importSecret(bytes const& _s, string const& _pass)
 {
-	h128 r = h128::random();
-	m_cached[r] = _s;
-	m_keys[r] = make_pair(encrypt(_s, _pass), std::string());
-	save();
+	h128 r;
```

Comment:
not better to throw in case difficulty to encrypt rather than return a zero uuid?